### DISCUSSION TITLE
How could I send an array of list as query parameters via POST method?

### DISCUSSION THREAD

**Discussion by laurence-lin** [fastapi/fastapi#6629] [Questions] [answered]
I have to send a list or array as parameters with POST method, like this:

```
@app.post('endpoint')

async def feedback(feed: List[int])
```

I expect to be able to send parameters of a list of integer, and return the response.

However, in the UI document, the endpoint couldn't receive request parameters, instead only showing return 0.

I'd looked for request body, and tried to do something like this:

```
class Item(BaseModel):
  feedback: int

async def feedback(feed: List[Item])
```

But that returns this: `feedback: 0`   and I still could not send parameters of a list.

How could I fulfill this task?

**Comment by hard-coders** [comment, marked as answer]
Normally, when you omit a type of parameter, FastAPI handles it as a query or path parameter.
When using POST method is going to be weird, however.
In this case, use `Query` class explicitly.

```python
from typing import List
from fastapi import FastAPI, Query

app = FastAPI()


@app.post("/endpoint")
async def feedback(feed: List[int] = Query(...)):
    return feed
```

Then you can see input tags in the docs.

**Comment by laurence-lin** [comment]
> Normally, when you omit a type of parameter, FastAPI handles it as a query or path parameter.
> When using POST method is going to be weird, however.
> In this case, use `Query` class explicitly.
> 
> ```python
> from typing import List
> from fastapi import FastAPI, Query
> 
> app = FastAPI()
> 
> 
> @app.post("/endpoint")
> async def feedback(feed: List[int] = Query(...)):
>     return feed
> ```
> 
> Then you can see input tags in the docs.

I intend to do it with query parameter, thank you! I've success to give a list of integer. 

But I still  have a small problem, that I have to type integer one by one, and in the curl command it shows like:

`http://my_api/endpoint/?parameter1 = value1&parameter2 = value2`

Is there any way I could input a list once like:

`http://my_api/endpoint/?parameter1 = {value1, value2, value3...}` ?

Thank you for help anyway.

**Comment by hard-coders** [comment]
Unfortunately, that's not possible.

But there is two way. One is using JSON body, and the other is that you need to parse a parameter yourself.

The former is a good way and recommended.
The latter looks ugly and not a good way.
```python
@app.post("/endpoint")
async def feedback(feed: List[str] = Query(...)):
    feed_list = [int(_) for _ in feed.split(",")]
    return feed_list
```
When using the latter, it can't appear your intention in docs.
I hope it helps.

**Comment by laurence-lin** [comment]
> Unfortunately, that's not possible.
> 
> But there is two way. One is using JSON body, and the other is that you need to parse a parameter yourself.
> 
> The former is a good way and recommended.
> The latter looks ugly and not a good way.
> 
> ```python
> @app.post("/endpoint")
> async def feedback(feed: List[str] = Query(...)):
>     feed_list = [int(_) for _ in feed.split(",")]
>     return feed_list
> ```
> 
> When using the latter, it can't appear your intention in docs.
> I hope it helps.

I understand, I may have to create an JSON file containing the list of values, and attach in the request body. Thank you!

**Comment by tiangolo** [comment]
Thanks for the help here @hard-coders ! 👏  🙇 

Thanks for reporting back and closing the issue @laurence-lin  👍

> Sorry for the long delay! 🙈  I wanted to personally address each issue/PR and they piled up through time, but now I'm checking each one in order.
